src/NServiceBus.Testing/Sagas/TestableSaga.cs

```diff
@@ -306,9 +303,29 @@ async Task<HandleResult> HandleTimeout(OutgoingMessage<object, SendOptions> time
                 return new SagaLoadResult { DiscardMessage = true };
             }
 
-            var messageMappedValue = sagaMapper.GetMessageMappedValue(message);
-
-            sagaData = await persister.Get<TSagaData>(sagaMapper.CorrelationPropertyName, messageMappedValue, session, context.Extensions, context.CancellationToken).ConfigureAwait(false);
+            var finder = FindSagaFinder(message.Type);
+            if (finder != null)
+            {
+                var findByMethod = finder.GetType().GetMethod("FindBy",
```

Comment:
If we're in full control of the actual IFindSaga classes being passed to saga metadata, why do we still need to lean on reflection?